extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/discovery/WeatherUndergroundDiscoveryService.java

```diff
@@ -0,0 +1,110 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.weatherunderground.discovery;
+
+import static org.eclipse.smarthome.binding.weatherunderground.WeatherUndergroundBindingConstants.*;
+import static org.eclipse.smarthome.binding.weatherunderground.internal.config.WeatherUndergroundConfiguration.*;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.i18n.LocationProvider;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WeatherUndergroundDiscoveryService} creates things based on the configured location.
+ *
+ * @author Laurent Garnier - Initial Contribution
+ */
+public class WeatherUndergroundDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherUndergroundDiscoveryService.class);
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_WEATHER);
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
+
+    private LocationProvider locationProvider;
+    private ScheduledFuture<?> discoveryJob;
+    private PointType previousLocation;
+
+    /**
+     * Creates a WeatherUndergroundDiscoveryService with enabled autostart.
+     */
+    public WeatherUndergroundDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, DISCOVER_TIMEOUT_SECONDS, true);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Weather Underground discovery scan");
+        PointType location = locationProvider.getLocation();
+        if (location == null) {
+            logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
+            return;
+        }
+        createResults(location);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (discoveryJob == null) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(() -> {
+                PointType currentLocation = locationProvider.getLocation();
+                if (!Objects.equals(currentLocation, previousLocation)) {
+                    logger.debug("Location has been changed from {} to {}: Creating new discovery results",
+                            previousLocation, currentLocation);
+                    createResults(currentLocation);
```

Comment:
Only the discovery result is updated, not the thing irself. So I don't see any problem here.